FGEGraphics/GraphicsHelpers/FontSets/FontSet.cs

```diff
@@ -730,6 +734,145 @@ public RenderableText SplitAppropriately(string text, string baseColor, int maxX
+    public static List<EditableTextLine> SplitWordAppropriately(EditableTextLine word, float maxWidth)
+    {
+        List<EditableTextLine> result = [new()];
+        List<RenderableTextPart> parts = [.. word.Parts];
+        for (int i = 0; i < parts.Count; i++)
+        {
+            EditableTextLine lastWord = result.Last();
+            RenderableTextPart part = parts[i];
+            if (lastWord.Width + part.Width <= maxWidth)
+            {
+                lastWord.AddPart(part);
+                continue;
+            }
+            float lastWidth = 0;
+            for (int j = 1; j <= part.Text.Length; j++)
+            {
+                float width = part.Font.MeasureString(part.Text[..j]);
+                if (lastWord.Width + width > maxWidth)
+                {
+                    RenderableTextPart firstPart = part.Clone();
+                    firstPart.Text = part.Text[..(j - 1)];
+                    firstPart.Width = lastWidth;
+                    RenderableTextPart secondPart = part.Clone();
+                    secondPart.Text = part.Text[(j - 1)..];
+                    secondPart.Width = part.Width - lastWidth;
+                    lastWord.AddPart(firstPart);
+                    result.Add(new());
+                    parts.Insert(i + 1, secondPart);
+                    break;
+                }
+                lastWidth = width;
+            }
+        }
+        return result;
+    }
+
+    /// <summary>Splits a renderable text line at a maximum render width.</summary>
+    /// <param name="line">The line to split.</param>
+    /// <param name="maxWidth">The maximum render width.</param>
+    /// <param name="skippedIndices">A list of character indices ignored in the final result.</param>
+    /// <returns>The multiple-line renderable result.</returns>
+    public static RenderableText SplitLineAppropriately(RenderableTextLine line, float maxWidth, out List<int> skippedIndices)
+    {
+        skippedIndices = [];
+        if (line.Width < maxWidth)
+        {
+            return new(line);
+        }
+        int charIndex = 0;
+        float totalWidth = 0;
+        EditableTextLine currentLine = new(true);
+        List<RenderableTextLine> lines = [];
+        void BuildLine()
+        {
+            if (currentLine.Parts.Count > 0)
+            {
+                lines.Add(currentLine.ToRenderable());
+            }
+            currentLine = new(true);
+        }
+        List<EditableTextLine> words = SplitLineIntoWords(line);
+        for (int i = 0; i < words.Count; i++)
+        {
+            EditableTextLine word = words[i];
+            bool needsSplit = word.Width > maxWidth;
+            bool needsNewLine = currentLine.Width + word.Width > maxWidth;
+            if (needsSplit || needsNewLine)
+            {
+                BuildLine();
+                if (word.IsWhitespace)
+                {
+                    skippedIndices.Add(++charIndex);
```

Comment:
A list of character indices... set based on words (?), as a linear potentially-dense listing of integers... this seems like a weird approach. I don't entirely follow what this is for, but are you sure this is the right way to do whatever this is meant for?

Reply:
Conceptually it should be a map of indices to strings (or just their lengths) but every 'whitespace' word is just a single space character